### ISSUE TITLE
Core Data: Multithreading violation when retrieving reminder information

### ISSUE THREAD

**Issue by staskus** [wordpress-mobile/WordPress-iOS#20842]
Core Data: Multithreading violation when setting blogging reminders.

Could be the cause of crashes in the production: https://github.com/wordpress-mobile/WordPress-iOS/issues/20840

### Steps to reproduce the behavior

1. Set blogging reminders (as described in https://github.com/wordpress-mobile/WordPress-iOS/issues/20841)
2. Pass `-com.apple.CoreData.ConcurrencyDebug 1` arguments on launch
3. Open Jetpack app and log in
4. Observe Multithreading_Violation

It happens when `reschedulePromptRemindersIfNeeded` is triggered which checks for the notification settings and then accesses `BloggingPromptSettingsReminderDays.getActiveWeekdays` from a different thread

```
#0	0x00000001845323a0 in +[NSManagedObjectContext __Multithreading_Violation_AllThatIsLeftToUsIsHonor__] ()
#1	0x0000000184531f7c in _PFAssertSafeMultiThreadedAccess_impl ()
#2	0x0000000184530f6c in _sharedIMPL_pvfk_core_c ()
#3	0x0000000105f1b4f0 in BloggingPromptSettingsReminderDays.getActiveWeekdays() at /Users/povilasstaskus/Projects/WordPress-iOS/WordPress/Classes/Models/BloggingPromptSettingsReminderDays+CoreDataClass.swift:19
#4	0x0000000105556148 in closure #1 in BloggingPromptCoordinator.reschedulePromptRemindersIfNeeded(for:completion:) at /Users/povilasstaskus/Projects/WordPress-iOS/WordPress/Classes/ViewRelated/Blog/Blogging Prompts/BloggingPromptCoordinator.swift:129
#5	0x0000000105556328 in thunk for @escaping @callee_guaranteed (@guaranteed UNNotificationSettings) -> () ()

```

https://github.com/wordpress-mobile/WordPress-iOS/assets/4062343/e7ccbd87-ca76-43ec-a3da-1624bd76a859

**Comment by dvdchr** [comment]
Closed via #20854

**Comment by dvdchr** [comment]
Updating priority to High, because I'm not sure if this is the actual cause for Matt's crashes.
